secretservice/dh_ietf1024_sha256_aes128_cbc_pkcs7.go

```diff
@@ -0,0 +1,145 @@
+// This file implements a basic Diffie-Hellman for groups with modular
+// exponentiation operators. In particular, it is used in this package
+// to implement the Diffie-Hellman KEX over the Second Oakley Group.
+// meant only for use for securing the channel to the D-Bus Secret Service.
+// Much of the code in this file is derived from package
+// golang.org/x/crypto/ssh:kex.go, and is replicated here because the relevant
+// variables and methods are not exported or easily accessible.
+// Note that this protocol is NOT authenticated, NOT secure against malleation
+// and is NOT CCA2-secure. It is only meant to hide the D-Bus messages from any
+// system services that may be logging everything.
```

Comment:
Could you add some unit tests for these functions?  If a lot of this is borrowed from crypto/ssh, are there tests from that package that we can also borrow?